client/blocks/reader-full-post/header.jsx

```diff
@@ -25,9 +31,30 @@ const ReaderFullPostHeader = ( { post, authorProfile } ) => {
 		return <ReaderFullPostHeaderPlaceholder />;
 	}
 
+	// Rather than pass in additional props for the `recent` layout, we extract the props we need from authorProfile.
+	const { props: { author, siteIcon, feedIcon, siteName, followCount } = {} } = authorProfile || {};
+
+	const isDefaultLayout = layout !== 'recent';
```

Comment:
Could `isDefaultLayout` be based on an empty `layout` prop, instead?

Reply:
Oh, good point! I think we can check if `layout === 'default';` because we set it as the "default" value [here](https://github.com/Automattic/wp-calypso/pull/96072/files#diff-dd39980f34de692faf2946104e4a84130d9defff05679cddf0a4049ee3306806R134). What do you think?

Reply:
Thinking about how I would type this in TS, I would probably do something like `layout: 'default' | 'recent'` to keep things explicit. Then I would set the default param (prop) to `'default'` so I could check against that value, i.e. `const isDefaultLayout = layout === 'default'` or go the other route and check for `'recent'`.

Just my two cents favoring explicit options over inferred ones.

Reply:
Oh, it looks like PropTypes.defaultProps injects the default values, so my suggestion would maybe use to positive check `===` instead of the negative `!==`.

Reply:
That works! 🙌